src/Board.py

```diff
@@ -10,28 +14,46 @@
 
 
 class Board:
-    def __init__(self, style="default", size=5) -> None:
-        extention = 0  # provision for games with more players
-        if size > 5:
-            extention = math.floor((math.ceil((size - 5) / 3) * 3) * 2.8)
-        if style == "random":
-            self.data = [
-                [
-                    Tile(terrain=random.randint(1, 9))
-                    for _ in range(BOARD_WIDTH + extention)
-                ]
-                for _ in range(BOARD_HEIGHT)
-            ]
-            self.data[2][-1] = 0
-            self.data[4][-1] = 0
-            self.data[6][-1] = 0
-            self.data[8][-1] = 0
-        elif style == "default":
+    priest_slots: Dict[Cult, int] = {
+        Cult.FIRE: 4,
+        Cult.EARTH: 4,
+        Cult.WATER: 4,
+        Cult.AIR: 4,
+    }
+    data: List[List[Tile]]
+
+    def __init__(self, style: str = "default", size: int = 5) -> None:
+        # Commented out the following because it doesn't work with strict typechecking
+        # extension = 0  # provision for games with more players
+        # if size > 5:
+        #     extension = math.floor((math.ceil((size - 5) / 3) * 3) * 2.8)
+        # if style == "random":
+        #     self.data = [
+        #         [
+        #             Tile(terrain=random.randint(1, 9))
```

Comment:
For context, this line is an issue with strict type checking because it uses integers instead of `Terrain` values. This is actually significant in the real world as well as in the type-checkers head because if, at some point, we change the terrain values to be different numbers or even change them to be strings, this would suddenly break.